File: LeetCode/Medium/2530-maximal-score-after-applying-k-operations/2530-maximal-score-after-applying-k-operations.cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        ios_base::sync_with_stdio(false);
        cin.tie(NULL);
        cout.tie(NULL);
        
        priority_queue<int> maxHeap(nums.begin(), nums.end());
        
        long ret = 0;

        for (int i = 0; i < k; i++) {
            int maxValue = maxHeap.top();
            maxHeap.pop();
            ret += maxValue;
            maxHeap.push((maxValue + 2) / 3);
        }

        return ret;
    }
};

File: LeetCode/Medium/2530-maximal-score-after-applying-k-operations/2530-maximal-score-after-applying-k-operations.java
class Solution {
    public long maxKelements(int[] nums, int k) {
        PriorityQueue<Integer> maxHeap = new PriorityQueue<>((a, b) -> b - a);

        for (int num : nums) {
            maxHeap.offer(num);
        }

        long ret = 0;

        for (int i = 0; i < k; i++) {
            int maxValue = maxHeap.poll();
            ret += maxValue;
            maxHeap.offer((maxValue + 2) / 3);
        }

        return ret;
    }
}